homeassistant/components/logbook/helpers.py

```diff
@@ -254,9 +252,16 @@ def _is_entity_id_filtered(
     Used to setup listeners and which entities to select
     from the database when a list of entities is requested.
     """
-    return bool(
-        split_entity_id(entity_id)[0] in ALWAYS_CONTINUOUS_DOMAINS
-        or (state := hass.states.get(entity_id))
-        and (ATTR_UNIT_OF_MEASUREMENT in state.attributes)
-        or is_sensor_continuous(ent_reg, entity_id)
-    )
+    # If its in continuous domains, we always want it
+    # filter it
+    if split_entity_id(entity_id)[0] in ALWAYS_CONTINUOUS_DOMAINS:
+        return True
+    # If it its in the state machine we can quick check if it
+    # has a unit_of_measurement or state_class, we want it
+    # filter it
+    if (state := hass.states.get(entity_id)) and (attributes := state.attributes):
+        return ATTR_UNIT_OF_MEASUREMENT in attributes or ATTR_STATE_CLASS in attributes
+    # If its not in the state machine, we need to check
+    # the entity registry to see if its a sensor
+    # filter with a state class
+    return is_sensor_continuous(ent_reg, entity_id)
```

Comment:
It was a bit redundant here to check the registry if the state is in the state machine